### DISCUSSION TITLE
ROS3 driver, presigned link support

### DISCUSSION THREAD

**Discussion by stephanlphilips** [HDFGroup/hdf5#4207] [Proposed Change]
**Is your feature request related to a problem?**
When testing the ROS3 driver, it worked well for files that are either fully public or where you have authentication credentials of the S3 account. However, when attempting to access via a pre-signed URL, I encountered an error indicating that curl could not process the request.

When looking in the source code I noticed that a placeholder for the query in the URL was left in the H5FD_s3comms_aws_canonical_request function in H5FDs3comms.c.

Is it correct to assume that this functionality is not available in the current implementation?

**Describe the solution you'd like**
I would like to see support for queries in the S3 URL. 
Are there other solutions I could consider that are compatible with Linux, macOS, and Windows?

**Describe alternatives you've considered**
One potential alternative could be using s3backer, although I'm unsure if this solution is compatible with Windows."

**Comment by ajelenak** [comment]
If you can try ROS3 from the library release 2.0 or later. Post here if it still does not work for you.
